src/business-logic/services/ingest-data-changed-service.ts

```diff
@@ -275,6 +299,23 @@ export class IngestDataChangedService implements DataChangeService {
     return events?.shift()
   }
 
+  private async generateActions(): Promise<void> {
+    for (const rundownId of this.rundownIdsToGenerateActionsFor) {
+      await this.generateActionsForRundown(rundownId)
+    }
+    this.rundownIdsToGenerateActionsFor.clear()
```

Comment:
We should note that the sequential nature of this can become an issue, if there is at least one bad rundown.

An alternative could be to await a `Promise.allSettled` and remove the ids of the settled ones. E.g.:
```typescript
    await Promise.allSettled(
      [...this.rundownIdsToGenerateActionsFor].map(async (rundownId) => {
        await this.generateActionsForRundown(rundownId)
        this.rundownIdsToGenerateActionsFor.delete(rundownId)
      })
    )
```
This way the rundowns that can generate actions will still be generated even if there is a bad rundown in the set.

Reply:
Updated